config/main.py

```diff
@@ -1045,20 +1138,98 @@ def add(session_name, src_ip, dst_ip, dscp, ttl, gre_type, queue, policer):
 
     if queue is not None:
         session_info['queue'] = queue
-    
+
+    if src_port is not None:
+        if get_interface_naming_mode() == "alias":
+            src_port_list = []
+            for port in src_port.split(","):
+                src_port_list.append(interface_alias_to_name(port))
+            src_port=",".join(src_port_list)
+
+        session_info['src_port'] = src_port
+        if direction is None:
+            direction = "both"
+
+    if direction is not None:
+        session_info['direction'] = direction.upper()
+
     """
     For multi-npu platforms we need to program all front asic namespaces
     """
     namespaces = sonic_device_util.get_all_namespaces()
     if not namespaces['front_ns']:
         config_db = ConfigDBConnector()
         config_db.connect()
+        if validate_mirror_session_config(config_db, session_name, None, src_port, direction) is False:
+            return
+        config_db.set_entry("MIRROR_SESSION", session_name, session_info)
+    else:
+        per_npu_configdb = {}
+        for front_asic_namespaces in namespaces['front_ns']:
+            per_npu_configdb[front_asic_namespaces] = ConfigDBConnector(use_unix_socket_path=True, namespace=front_asic_namespaces)
+            per_npu_configdb[front_asic_namespaces].connect()
+            if validate_mirror_session_config(per_npu_configdb[front_asic_namespaces], session_name, None, src_port, direction) is False:
+                return
+            per_npu_configdb[front_asic_namespaces].set_entry("MIRROR_SESSION", session_name, session_info)
+
+@add.command('span')
+@click.argument('session_name', metavar='<session_name>', required=True)
+@click.argument('dst_port', metavar='<dst_port>', required=True)
+@click.argument('src_port', metavar='[src_port]', required=False)
+@click.argument('direction', metavar='[direction]', required=False)
+@click.argument('queue', metavar='[queue]', required=False)
+@click.option('--policer')
+def span(session_name, dst_port, src_port, direction, queue, policer):
+    """
+    Add port mirror session
+    """
+    if get_interface_naming_mode() == "alias":
+        dst_port = interface_alias_to_name(dst_port)
+        if dst_port is None:
+            click.echo("Error: Destination Interface {} is invalid".format(dst_port))
+            return
+        if src_port is not None:
+            src_port_list = []
+            for port in src_port.split(","):
+                src_port_list.append(interface_alias_to_name(port))
+            src_port=",".join(src_port_list)
+
+    session_info = {
+            "type" : "SPAN",
+            "dst_port": dst_port,
+            }
+
+    if src_port is not None:
+        session_info['src_port'] = src_port
+        if direction is None:
+            direction = "both"
+
+    if direction is not None:
+        session_info['direction'] = direction.upper()
+
+    if policer is not None:
+        session_info['policer'] = policer
```

Comment:
What is the need for policer and queue for SPAN sessions?

Reply:
This is existing support in community. This is to police the mirrored traffic.